platform/mbed_error.c

```diff
@@ -216,7 +216,7 @@ mbed_error_status_t mbed_error_initialize(void)
             mbed_error_reboot_callback(report_error_ctx);
 
             //We let the callback reset the error info, so check if its still valid and do the rest only if its still valid.
-            if (report_error_ctx->error_reboot_count > 0) {
+            if (report_error_ctx->error_reboot_count < 0) {
```

Comment:
I think this essentially removes the bug fix we originally did. I think we should not commit this and focus on #9544.